### PR TITLE
feat(cli): detect non-Node repo validation commands

### LINKED ISSUE #235: feat(cli): detect non-Node repo validation commands during workflow init
## Background

`gh-symphony workflow init` currently generates repository-aware guidance from `detectEnvironment()`, but the detector is still heavily Node-centric:

- package manager detection is based on JS lockfiles
- validation guidance comes from `package.json` `test` / `lint` / `build` scripts
- monorepo detection is mostly JS workspace signals

That works well for TypeScript repos, but it degrades quickly for Python, Go, Rust, Make-based, or mixed-language repositories. In those cases the generated `WORKFLOW.md` falls back to generic guidance even when the repository already exposes clear validation entry points.

For adoption, this is a product gap: GitHub Symphony positions itself as a repository-driven orchestration layer, so the out-of-the-box setup should feel credible beyond Node projects.

## Proposed Changes

1. Extend `detectEnvironment()` to inspect common non-Node signals such as:
   - Python: `pyproject.toml`, `uv.lock`, `poetry.lock`, `requirements*.txt`, `pytest.ini`, `tox.ini`
   - Go: `go.mod`
   - Rust: `Cargo.toml`
   - Generic command runners: `Makefile`, `justfile`
2. Derive conservative repository validation commands when the signals are unambiguous, for example:
   - `make test`, `make lint`, `make build`
   - `just test`, `just lint`, `just build`
   - `uv run pytest`, `poetry run pytest`, or `pytest`
   - `go test ./...`
   - `cargo test`
3. Thread those detected commands into the existing repository guidance generator so `WORKFLOW.md`, `.gh-symphony/reference-workflow.md`, and generated skill files stay aligned.
4. Keep the current generic fallback when detection is ambiguous. The detector should prefer explicit repository entry points over clever guessing.
5. Add fixture-based tests for at least one Python, Go, Rust, and Make-based repository shape.
6. Document the broader detector support in README so the CLI feels language-agnostic rather than Node-only.

## Affected Files

| File | Change | Scope |
| --- | --- | --- |
| `packages/cli/src/detection/environment-detector.ts` | extend ecosystem detection beyond `package.json` | CLI generation |
| `packages/cli/src/workflow/repository-guidance.ts` | support rendering detected non-Node commands cleanly | Prompt generation |
| `packages/cli/src/commands/init.ts` | pass richer detection output through the existing scaffold flow | CLI UX |
| `packages/cli/src/workflow/generate-reference-workflow.ts` | keep reference workflow guidance aligned | Docs generation |
| `packages/cli/src/skills/templates/*.ts` | preserve consistent repo-aware validation guidance in generated skills | Skill generation |
| `packages/cli/src/detection/environment-detector.test.ts` | add multi-ecosystem fixtures and regressions | Verification |
| `README.md` | document supported ecosystem detection | Adoption |
| `packages/cli/README.md` | add examples and expectations for non-Node repos | Adoption |

## Guardrails

- Never invent destructive commands.
- Prefer explicit repository entry points (`Makefile`, `justfile`, existing tool config) over inferred tool invocations.
- If multiple signals conflict, fall back to generic guidance instead of guessing wrong.

## Why This Is High-Value

- Utility: high — it improves scaffold quality for a much larger set of repositories.
- Size: medium — the work is localized to detection/generation/tests, not core orchestration.
- Product impact: strong — it makes GitHub Symphony feel genuinely repo-agnostic, which broadens adoption and improves first-run quality.

## References

- `packages/cli/src/detection/environment-detector.ts`
- `packages/cli/src/workflow/repository-guidance.ts`
- `packages/cli/src/workflow/generate-workflow-md.ts`

### PR DESCRIPTION
**Pull request by moncher-dev**
## Issues

- Fixes #235

## Summary

- extend CLI repository detection beyond Node-specific `package.json` scripts and lockfiles
- generate conservative validation guidance for Python, Go, Rust, Make, and Just based repositories during `workflow init`
- tighten justfile detection so variable assignments do not masquerade as runnable recipes

## Changes

- add non-Node detector support for `uv.lock`, `poetry.lock`, `pyproject.toml`, `pytest.ini`, `requirements*.txt`, `go.mod`, `Cargo.toml`, `Makefile`, and `justfile`
- resolve validation commands by confidence so explicit `make`/`just` entry points win over package scripts, and same-priority conflicts fall back to generic guidance
- distinguish just recipe declarations from `:=` assignments and add regression coverage for variable-only justfiles
- detect `go.work` and Cargo workspaces as additional monorepo signals
- add fixture coverage for Python, Go, Rust, Make, monorepo, conflicting command-runner cases, and the justfile assignment edge case
- update the root and CLI READMEs to describe language-agnostic validation command detection in `workflow init`

## Evidence

- `pnpm --filter @gh-symphony/cli test -- src/detection/environment-detector.test.ts`
- `pnpm lint`
- `pnpm test`
- `pnpm typecheck`
- `pnpm build`
- Manual check: re-read issue #235 against the final diff and confirmed the implementation plus follow-up review fix cover non-Node signal detection, conservative command derivation, threaded workflow guidance, fixture coverage, README updates, and the justfile assignment regression

## Human Validation

- [ ] Confirm `gh-symphony workflow init --dry-run` surfaces non-Node validation guidance for representative Python, Go, Rust, and Make repositories
- [ ] Confirm generated `WORKFLOW.md`, `.gh-symphony/reference-workflow.md`, and skill files stay aligned with the detected repository commands
- [ ] Confirm mixed or conflicting repository signals still fall back to generic validation guidance instead of guessing
- [ ] Confirm justfiles that only define `test := ...` style variables do not emit `just test` guidance

## Risks

- Python command derivation intentionally stays conservative and currently only emits `pytest`-based guidance when pytest configuration is detectable, so repositories that rely on other Python runners still fall back to generic guidance

### CODE REVIEW

**Review by hojinzs — COMMENTED** [review]
## 리뷰 결과

**CI 상태**: Test ✅, Container Smoke ✅ — 모두 통과

---

### P2 버그 수정 (리뷰 코멘트 반영)

`detectJustCommands`의 `^${name}\\s*:` 정규식이 justfile 변수 할당(`test := "cargo test"`)도 레시피로 잘못 감지하는 문제를 수정했습니다.

`:=` 앞의 `:` 에도 매칭되어 레시피가 없는 경우에도 `just test` 같은 유효하지 않은 커맨드가 생성될 수 있었습니다.

**수정 내용**: 네거티브 룩어헤드 `(?!=)` 추가

```typescript
// 수정 전
new RegExp(`^${name}\\s*:`, "m").test(justfile)

// 수정 후
new RegExp(`^${name}\\s*:(?!=)`, "m").test(justfile)
```

변수 할당만 있고 레시피가 없는 justfile에 대한 회귀 테스트도 추가했습니다. 28개 테스트 모두 통과 확인 (환경 감지 관련), typecheck/lint 이상 없음.

> 참고: `doctor.test.ts` 6개 실패는 이 PR과 무관하게 머지 전부터 존재하는 기존 문제입니다.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/cli/src/detection/environment-detector.ts [hunk 1]
```diff
@@ -90,6 +162,156 @@ async function detectScripts(cwd: string): Promise<{
...
+async function detectJustCommands(cwd: string): Promise<DetectedCommands> {
+  const justfile =
+    (await readTextFile(join(cwd, "justfile"))) ??
+    (await readTextFile(join(cwd, ".justfile")));
+  if (!justfile) {
+    return { testCommand: null, buildCommand: null, lintCommand: null };
+  }
+
+  const hasRecipe = (name: string): boolean =>
+    new RegExp(`^${name}\\s*:`, "m").test(justfile);
```
[the remark this answers is not in the document]
Comment (moncher-dev) [reply packages/cli/src/detection/environment-detector.ts]: Fixed in `9bd7ff9`. `detectJustCommands` now matches `^<name>\s*:(?!=)` so `test := "cargo test"` style assignments no longer count as recipes, and I added a regression test that verifies variable-only justfiles leave `testCommand`, `lintCommand`, and `buildCommand` unset.
